test/unittests/test_cli.py

```diff
@@ -0,0 +1,157 @@
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "https://api.dasch.swiss": [
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "https://app.dasch.swiss": [
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "https://iiif.dasch.swiss": [
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "https://dasch.swiss": [
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "dasch.swiss": [
+                "https://api.dasch.swiss",
+                "https://iiif.dasch.swiss",
+            ],
+            "http://admin.test.dasch.swiss/": [
+                "https://api.test.dasch.swiss",
+                "https://iiif.test.dasch.swiss",
+            ],
+            "http://app.staging.dasch.swiss/": [
+                "https://api.staging.dasch.swiss",
+                "https://iiif.staging.dasch.swiss",
+            ],
+            "https://demo.dasch.swiss/": [
+                "https://api.demo.dasch.swiss",
+                "https://iiif.demo.dasch.swiss",
+            ],
+            "http://api.dev.dasch.swiss/": [
+                "https://api.dev.dasch.swiss",
+                "https://iiif.dev.dasch.swiss",
+            ],
+            "dev-02.dasch.swiss": [
+                "https://api.dev-02.dasch.swiss",
+                "https://iiif.dev-02.dasch.swiss",
+            ],
+            "082a-test-server.dasch.swiss": [
+                "https://api.082a-test-server.dasch.swiss",
+                "https://iiif.082a-test-server.dasch.swiss",
+            ],
+            "admin.08F4-test-server.dasch.swiss": [
+                "https://api.08F4-test-server.dasch.swiss",
+                "https://iiif.08F4-test-server.dasch.swiss",
+            ],
+            "app.08F4-test-server.dasch.swiss": [
+                "https://api.08F4-test-server.dasch.swiss",
+                "https://iiif.08F4-test-server.dasch.swiss",
+            ],
+            "iiif.E5bC-test-server.dasch.swiss": [
+                "https://api.E5bC-test-server.dasch.swiss",
+                "https://iiif.E5bC-test-server.dasch.swiss",
+            ],
+            "not-yet-0826-test-server.dasch.swiss": [
+                "https://api.not-yet-0826-test-server.dasch.swiss",
+                "https://iiif.not-yet-0826-test-server.dasch.swiss",
+            ],
+            "https://admin.ls-prod-server.dasch.swiss": [
+                "https://api.ls-prod-server.dasch.swiss",
+                "https://iiif.ls-prod-server.dasch.swiss",
+            ],
+            "https://ls-test-server.dasch.swiss": [
+                "https://api.ls-test-server.dasch.swiss",
+                "https://iiif.ls-test-server.dasch.swiss",
+            ],
+        }
```

Comment:
I'm not a fan of this kind of test parameterisation: if you want to test arbitrary strings because you're concerned the behaviour might be flaky, then you should use a framework for property-based testing (like Hypothesis); otherwise you should identify the different possible code paths and test those systematically with one test per possible case.  
If you expect a different behaviour for `"082a-test-server.dasch.swiss"` and `"http://api.dev.dasch.swiss/"` then this test either tests the regex you wrote (if you don't trust that) or the correctness of the python regex implementation. If the first is the case, then factor out your regexes and test them separately, if the latter is the case, then don't! 😉 

And if you reduce the number of cases you want to test, then you don't need to parametrise your tests, and instead you can have a dedicated test method for each case you want to test, which gives you clearer and more expressive test results.

Reply:
I think there is another case: I'd like my unit tests to be an executable requirements specification. I think I have read that in David Farley's "Modern Software Engineering", but I'm not sure.

The idea that I have in mind: Some day, someone might come across and modify my code, so that by mistake, one of the subdomains isn't covered any more. So I want my unit tests to be a protection against future harm. 

Does that make sense?

Reply:
Okay - I see your point, and will keep it in mind. thanks for the input, it's always interesting to exchange thoughts with you, and I can learn a lot from your comments :-)